pkg/apis/softwarecomposition/v1beta1/types.go

```diff
@@ -560,6 +566,25 @@ type Syscall struct {
 	Args []*Arg `json:"args,omitempty" protobuf:"bytes,4,rep,name=args"`
 }
 
+type HTTPEndpoint struct {
+	Endpoint  string                  `json:"endpoint,omitempty" protobuf:"bytes,1,opt,name=endpoint"`
+	Methods   []string                `json:"methods,omitempty" protobuf:"bytes,2,opt,name=methods"`
+	Internal  bool                    `json:"internal" protobuf:"bytes,3,opt,name=internal"`
+	Direction consts.NetworkDirection `json:"direction,omitempty" protobuf:"bytes,4,opt,name=direction"`
+	Headers   json.RawMessage         `json:"headers,omitempty" protobuf:"bytes,5,opt,name=headers"`
+}
+
+func (e *HTTPEndpoint) GetHeaders() (map[string][]string, error) {
```

Comment:
I am not sure we need the GetHeaders in both types... (especially the other one, since the unversioned type should only be used for internal storage stuff)

Reply:
I need it for node agent